From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Andrew Branson <andrew.branson@jolla.com>
Date: Fri, 7 Aug 2026 11:08:52 +0200
Subject: [sailfishos][sqlite] Export system SQLite boolean substitution

Expose the package-detection result as a boolean so AppConstants can be preprocessed.
---
 build/moz.configure/sqlite.configure | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/build/moz.configure/sqlite.configure b/build/moz.configure/sqlite.configure
index 2a15bff25f1b1..6650cc450645f 100644
--- a/build/moz.configure/sqlite.configure
+++ b/build/moz.configure/sqlite.configure
@@ -14,5 +14,7 @@ system_sqlite = pkg_check_modules(
     "SQLITE", "sqlite3 >= 3.49.2", when="--with-system-sqlite"
 )
 
-set_config("MOZ_SYSTEM_SQLITE", True, when=system_sqlite)
-set_define("MOZ_SYSTEM_SQLITE", True, when=system_sqlite)
+system_sqlite_enabled = depends_if(system_sqlite)(lambda _: True)
+
+set_config("MOZ_SYSTEM_SQLITE", system_sqlite_enabled)
+set_define("MOZ_SYSTEM_SQLITE", system_sqlite_enabled)
